src/vs/workbench/browser/parts/editor/editorPart.ts

```diff
@@ -709,15 +711,31 @@ export class EditorPart extends Part implements EditorGroupsServiceImpl, IEditor
 		addClass(this.container, 'content');
 		parent.appendChild(this.container);
 
-		// Grid control
+		// Grid control with center layout
 		this.doCreateGridControl();
+		this.centeredViewLayout = new CenteredViewLayout(this.container, {
+			element: this.gridWidget.element,
+			layout: size => this.gridWidget.layout(size, this.dimension ? this.dimension.height : this.gridWidget.maximumHeight),
```

Comment:
This looks super weird. There must always be a `this.dimension`. It's impossible for this view to be asked for layout before `doLayout` has run.

Also, if you want to remove the weirdness, don't simply use `IView` from splitview. Use `IView` from gridview. This one talks about `width` and `height`. Simultaneously, `CenteredViewLayout.layout` would take in both `width` and `height`.